### PR TITLE
tests: test factories for User and related models INSPIR-427

### PR DESCRIPTION
**Pull request by puntonim**
https://its.cern.ch/jira/browse/INSPIR-427
https://its.cern.ch/jira/browse/INSPIR-334

### CODE REVIEW

File: tests/integration/orcid/__init__.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import imp
+import os
+
+
+# FIXME: Ugly hack to be able to import from the 'tests' module.
```
Comment (puntonim) [inline tests/integration/orcid/__init__.py]: This is very ugly and we will find a better solution in the next PRs.
The problem is that we badly named our test module.

Comment (puntonim) [reply tests/integration/orcid/__init__.py]: We decided to use pytest fixtures to solve this

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -45,6 +45,7 @@
 # Use the helpers folder to store test helpers.
 # See: http://stackoverflow.com/a/33515264/374865
 sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'helpers'))
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(__file__)), 'test_factory'))
```
Comment (puntonim) [inline tests/integration/conftest.py]: The test dir is not a module (for valid reasons...) thus we need this trick to add the test factory dir to the system path.

Comment (jacquerie) [reply tests/integration/conftest.py]: Alternative idea: put it in `tests/helpers/factories/db`.

Comment (puntonim) [reply tests/integration/conftest.py]: Yesss

File: tests/test_factory/test_factory_db/base.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    def _filter_kwargs_for_model(kwargs, model_class):
+        """
+        Filter the given kwargs and keep only those that make sense for
+        the given model_class.
+
+        Args:
+            kwargs (dict): a dictionary.
+            model_class: the DB model class to filter the kwargs for.
+
+        Returns (dict): a dict containing only those kwargs that make sense
```
Comment (jacquerie) [inline tests/test_factory/test_factory_db/base.py]: This yields a warning when building the docs, it should be:
```
Returns:
    dict: a dict containing only those kwargs that make sense
    for the given model_class.
```

File: tests/test_factory/test_factory_db/invenio_accounts.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+class TestUser(TestBaseModel):
+    """
+    Create User instances.
+
+    Basic usage:
+        >>> factory = TestUser.create_from_kwargs(email='foo@bar.com')
+        >>> factory.user
+        <User (transient 4661300240)>
+        >>> factory.user.email
+        'foo@bar.com'
```
Comment (jacquerie) [inline tests/test_factory/test_factory_db/invenio_accounts.py]: You can use `Examples:` for this, see https://sphinxcontrib-napoleon.readthedocs.io/en/latest/#docstring-sections.

File: tests/test_factory/test_factory_db/invenio_oauthclient.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    Create UserIdentity instances.
+
+    Basic usage:
+        >>> factory = TestUserIdentity.create_from_kwargs(id='myorcid')
+        >>> factory.user_identity
+        <UserIdentity (transient 4661300240)>
+        >>> factory.user_identity.id
+        'myorcid'
+        >>> factory.user
+        <User 41>
```
Comment (jacquerie) [inline tests/test_factory/test_factory_db/invenio_oauthclient.py]: Same as https://github.com/inspirehep/inspire-next/pull/3296/files#r178035094.

File: tests/test_factory/test_factory_db/invenio_oauthclient.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    Create RemoteAccount instances.
+
+    Basic usage:
+        >>> factory = TestRemoteAccount.create_from_kwargs(client_id='myclientid')
+        >>> factory.remote_account
+        Remote Account <id=9, user_id=42>
+        >>> factory.remote_account.client_id
+        'myclientid'
+        >>> factory.user
+        <User 41>
```
Comment (jacquerie) [inline tests/test_factory/test_factory_db/invenio_oauthclient.py]: Same as https://github.com/inspirehep/inspire-next/pull/3296/files#r178035094.

File: tests/test_factory/test_factory_db/invenio_oauthclient.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    Create RemoteToken instances.
+
+    Basic usage:
+        >>> factory = TestRemoteToken.create_from_kwargs(access_token='myaccesstoken')
+        >>> factory.remote_token
+        Remote Token <token_type=None access_token=****oken>
+        >>> factory.remote_token.access_token
+        'myaccesstoken'
+        >>> factory.remote_account
+        Remote Account <id=10, user_id=43>
```
Comment (jacquerie) [inline tests/test_factory/test_factory_db/invenio_oauthclient.py]: Same as https://github.com/inspirehep/inspire-next/pull/3296/files#r178035094.

File: tests/integration/conftest.py (hunk 2 above)
Comment (puntonim) [reply tests/integration/conftest.py]: done

File: tests/test_factory/test_factory_db/base.py (hunk 3 above)
Comment (puntonim) [reply tests/test_factory/test_factory_db/base.py]: done

File: tests/test_factory/test_factory_db/invenio_accounts.py (hunk 4 above)
Comment (puntonim) [reply tests/test_factory/test_factory_db/invenio_accounts.py]: done

File: tests/test_factory/test_factory_db/invenio_oauthclient.py (hunk 5 above)
Comment (puntonim) [reply tests/test_factory/test_factory_db/invenio_oauthclient.py]: done

File: tests/test_factory/test_factory_db/invenio_oauthclient.py (hunk 6 above)
Comment (puntonim) [reply tests/test_factory/test_factory_db/invenio_oauthclient.py]: done

File: tests/test_factory/test_factory_db/invenio_oauthclient.py (hunk 7 above)
Comment (puntonim) [reply tests/test_factory/test_factory_db/invenio_oauthclient.py]: done

File: tests/integration/helpers/factories/__init__.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (puntonim) [inline tests/integration/helpers/factories/__init__.py]: This file is required in order to be able to do:
`from factories.db.invenio_oauthclient import TestRemoteToken`
The same for the other init files.

File: inspirehep/modules/orcid/utils.py [hunk 9]
```diff
@@ -126,6 +126,19 @@ def get_push_access_token(orcid):
...
+    remote_accounts = db.session.query(RemoteAccount)\
+        .filter(RemoteAccount.user_id == UserIdentity.id_user)\
+        .filter(UserIdentity.id.in_(orcids)).all()
+
+    remote_tokens = []
+    for remote_account in remote_accounts:
+        if remote_account.extra_data.get('allow_push', False):
+            remote_tokens.extend(remote_account.remote_tokens)
+
+    return remote_tokens
```
[the remark this answers is not in the document]
Comment (puntonim) [reply inspirehep/modules/orcid/utils.py]: done

**Pull request merged**